### PR TITLE
Ensure that IF_*WR_RRD formats support 4-byte SIMD instructions

### LINKED ISSUE #85469: Test failure: _Avx512F_ro::JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16WithSaturationVector512UInt32()
coreclr windows x64 Checked jitstress2 @ Windows.10.Amd64.Open

```
06:07:39.227 Running test: _Avx512F_ro::JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16WithSaturationVector512UInt32()
Beginning scenario: RunBasicScenario_UnsafeRead
Beginning scenario: RunBasicScenario_Load
Beginning scenario: RunBasicScenario_LoadAligned
Beginning scenario: RunReflectionScenario_UnsafeRead
Beginning scenario: RunLclVarScenario_UnsafeRead
Beginning scenario: RunClassFldScenario
Beginning scenario: RunStructLclFldScenario
Avx512F.ConvertToVector256UInt16WithSaturation<UInt16>(Vector512<UInt32>): RunStructLclFldScenario failed:
  firstOp: (29684, 797567, 194292, 474623, 318946, 1007584, 646866, 88786, 784756, 707864, 550833, 515853, 449525, 982800, 1046206, 438200)
   result: (29684, 11135, 63220, 15871, 56802, 24544, 57042, 23250, 63860, 52504, 26545, 57101, 56309, 65296, 63166, 44984)

Beginning scenario: RunStructFldScenario
06:07:40.025 Failed test: _Avx512F_ro::JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16WithSaturationVector512UInt32()
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=254139&view=ms.vss-test-web.build-test-results-tab&runId=4942126&paneView=debug&resultId=103076

@tannergooding

### LINKED ISSUE #85586: Test failure _Avx512F_ro::JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16Vector512UInt32()
windows/linux x64 jitstress2-jitstressregs

```
DOTNET_TieredCompilation=0
DOTNET_JitStress=2
DOTNET_JitStressRegs=0x80
```

```
18:56:45.934 Running test: _Avx512F_ro::JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16WithSaturationVector512UInt32()
Beginning scenario: RunBasicScenario_UnsafeRead
Beginning scenario: RunBasicScenario_Load
Beginning scenario: RunBasicScenario_LoadAligned
Beginning scenario: RunReflectionScenario_UnsafeRead
Beginning scenario: RunLclVarScenario_UnsafeRead
Beginning scenario: RunClassFldScenario
Beginning scenario: RunStructLclFldScenario
Fatal error. System.Runtime.InteropServices.SEHException (0x80004005): External component has thrown an exception.
   at JIT.HardwareIntrinsics.X86._Avx512F.SimpleUnaryOpConvTest__ConvertToVector256UInt16WithSaturationVector512UInt32.RunStructLclFldScenario()
   at JIT.HardwareIntrinsics.X86._Avx512F.Program.ConvertToVector256UInt16WithSaturationVector512UInt32()
   at Program.<<Main>$>g__TestExecutor7|0_6(System.IO.StreamWriter, System.IO.StreamWriter, <>c__DisplayClass0_0 ByRef)
   at Program.<Main>$(System.String[])
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=257559&view=ms.vss-test-web.build-test-results-tab

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/runtime/issues/85586 and resolves https://github.com/dotnet/runtime/issues/85469

The general issue is that we have some scenarios where the instruction opcode is baselined at 4-bytes. In this case, we put the opcode in the `code` byte that `modr/m` is normally stored and so we can't pre-fill that with the regcode.

This fixes the formats that are new to supporting this scenario with EVEX to handle it correctly.

### CODE REVIEW

File: src/coreclr/jit/emitxarch.cpp [hunk 1]
```diff
@@ -16269,19 +16277,33 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
...
+                dst = emitOutputAM(dst, id, code);
+            }
+            else
+            {
+                code    = AddSimdPrefixIfNeeded(id, code, size);
+                regcode = (insEncodeReg345(id, id->idReg1(), size, &code) << 8);
+                dst     = emitOutputAM(dst, id, code | regcode);
+            }
+
+            sz = emitSizeOfInsDsc(id);
```
Comment (tannergooding) [inline src/coreclr/jit/emitxarch.cpp:16304]: There's actually a bit of code duplication that exists between these format specific handlers and the general `emitOutput**` methods.

I want to get that cleaned up and it should be easier once https://github.com/dotnet/runtime/pull/85536 goes in.

**Review by kunalspathak — APPROVED** [review]
LGTM when the CI is green.

**Review by BruceForstall — APPROVED** [review]

**Comment by tannergooding** [comment]
Fuzzlyn failure is pre-existing (at least back to .NET 8 P3): https://github.com/dotnet/runtime/issues/85611

**Comment by tannergooding** [comment]
Expecting a few JitStress failures due to https://github.com/dotnet/runtime/issues/85608
* `EnableHWIntrinsic=0`
* `EnableSSE=0`
* `EnableSSE2=0`

**Pull request merged**